1.17 error

## error
```
[01:06:17 ERROR]: Could not pass event BlockMultiPlaceEvent to Insights v6.4.0
java.lang.ExceptionInInitializerError: null
        at dev.frankheijden.insights.listeners.BlockListener.onBlockPlace(BlockListener.java:62) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor21.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:624) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at org.bukkit.craftbukkit.v1_17_R1.event.CraftEventFactory.callBlockMultiPlaceEvent(CraftEventFactory.java:360) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.world.item.ItemStack.placeItem(ItemStack.java:372) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.level.PlayerInteractManager.a(PlayerInteractManager.java:568) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1743) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.network.protocol.game.PacketPlayInUseItem.a(PacketPlayInUseItem.java:33) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.network.protocol.game.PacketPlayInUseItem.a(PacketPlayInUseItem.java:9) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$ensureRunningOnSameThread$1(PlayerConnectionUtils.java:36) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.util.thread.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:149) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.executeTask(IAsyncTaskHandlerReentrant.java:23) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.executeTask(MinecraftServer.java:1340) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.executeTask(MinecraftServer.java:193) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.util.thread.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:122) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.bg(MinecraftServer.java:1319) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.executeNext(MinecraftServer.java:1312) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.util.thread.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:132) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.sleepForTick(MinecraftServer.java:1273) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1184) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: dev.frankheijden.insights.dependencies.minecraftreflection.exceptions.MinecraftReflectionException: Invalid class 'net.minecraft.server.v1_17_R1.BlockPosition'!
        at dev.frankheijden.insights.dependencies.minecraftreflection.Reflection.getClassFromName(Reflection.java:40) ~[?:?]
        at dev.frankheijden.insights.dependencies.minecraftreflection.MinecraftReflection.of(MinecraftReflection.java:20) ~[?:?]
        at dev.frankheijden.insights.api.utils.BlockUtils.<clinit>(BlockUtils.java:20) ~[?:?]
        ... 26 more
Caused by: java.lang.ClassNotFoundException: net.minecraft.server.v1_17_R1.BlockPosition
        at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:146) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:519) ~[?:?]
        at java.lang.Class.forName0(Native Method) ~[?:?]
        at java.lang.Class.forName(Class.java:375) ~[?:?]
        at dev.frankheijden.insights.dependencies.minecraftreflection.Reflection.getClassFromName(Reflection.java:38) ~[?:?]
        at dev.frankheijden.insights.dependencies.minecraftreflection.MinecraftReflection.of(MinecraftReflection.java:20) ~[?:?]
        at dev.frankheijden.insights.api.utils.BlockUtils.<clinit>(BlockUtils.java:20) ~[?:?]
        ... 26 more
[01:06:20 ERROR]: Could not pass event BlockBreakEvent to Insights v6.4.0
java.lang.NoClassDefFoundError: Could not initialize class dev.frankheijden.insights.api.utils.BlockUtils
        at dev.frankheijden.insights.listeners.BlockListener.onBlockBreak(BlockListener.java:112) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor25.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:624) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.level.PlayerInteractManager.breakBlock(PlayerInteractManager.java:369) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.level.PlayerInteractManager.a(PlayerInteractManager.java:329) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.level.PlayerInteractManager.a(PlayerInteractManager.java:219) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1677) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.network.protocol.game.PacketPlayInBlockDig.a(PacketPlayInBlockDig.java:34) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.network.protocol.game.PacketPlayInBlockDig.a(PacketPlayInBlockDig.java:8) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$ensureRunningOnSameThread$1(PlayerConnectionUtils.java:36) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.util.thread.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:149) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.executeTask(IAsyncTaskHandlerReentrant.java:23) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.executeTask(MinecraftServer.java:1340) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.executeTask(MinecraftServer.java:193) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.util.thread.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:122) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.bg(MinecraftServer.java:1319) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.executeNext(MinecraftServer.java:1312) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.util.thread.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:132) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.sleepForTick(MinecraftServer.java:1273) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1184) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320) ~[patched_1.17.jar:git-Paper-"33a04d9"]
        at java.lang.Thread.run(Thread.java:831) [?:?]
```

## Actions to reproduce (if known)
Put Insight in the plugins folder and start it. place and destroy the bed.

## pluginlist
insights-6.4.0

## server version
paper-1.17-28.jar

--- a/Insights-API/src/main/java/dev/frankheijden/insights/api/utils/BlockUtils.java
+++ b/Insights-API/src/main/java/dev/frankheijden/insights/api/utils/BlockUtils.java
@@ -1,14 +1,15 @@
 package dev.frankheijden.insights.api.utils;
 
 import dev.frankheijden.insights.api.reflection.RTileEntityTypes;
-import dev.frankheijden.minecraftreflection.ClassObject;
 import dev.frankheijden.minecraftreflection.MinecraftReflection;
+import net.minecraft.core.BlockPosition;
+import net.minecraft.world.level.World;
+import net.minecraft.world.level.block.entity.TileEntity;
 import org.bukkit.Location;
 import org.bukkit.Material;
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
 import org.bukkit.block.BlockState;
-import org.bukkit.block.data.BlockData;
 import org.bukkit.block.data.type.Bed;
 import java.util.Optional;
 
@@ -16,16 +17,8 @@ public class BlockUtils {
 
     private static final MinecraftReflection craftBlockStateReflection = MinecraftReflection
             .of("org.bukkit.craftbukkit.%s.block.CraftBlockState");
-    private static final MinecraftReflection blockPositionReflection = MinecraftReflection
-            .of("net.minecraft.server.%s.BlockPosition");
     private static final MinecraftReflection craftWorldReflection = MinecraftReflection
             .of("org.bukkit.craftbukkit.%s.CraftWorld");
-    private static final MinecraftReflection worldReflection = MinecraftReflection
-            .of("net.minecraft.server.%s.World");
-    private static final MinecraftReflection nbtTagCompoundReflection = MinecraftReflection
-            .of("net.minecraft.server.%s.NBTTagCompound");
-    private static final MinecraftReflection tileEntityReflection = MinecraftReflection
-            .of("net.minecraft.server.%s.TileEntity");
 
     private BlockUtils() {}
 
@@ -45,28 +38,9 @@ public class BlockUtils {
         return craftBlockStateReflection.getClazz().isInstance(state);
     }
 
-    public static Object getTileEntity(Location loc) {
-        Object nmsWorld = craftWorldReflection.invoke(loc.getWorld(), "getHandle");
-        return getTileEntity(nmsWorld, createBlockPosition(loc));
-    }
-
-    public static Object getTileEntity(Object nmsWorld, Object blockPosition) {
-        return worldReflection.invoke(nmsWorld, "getTileEntity", blockPosition);
-    }
-
-    public static Object createBlockPosition(Location loc) {
-        return createBlockPosition(loc.getBlockX(), loc.getBlockY(), loc.getBlockZ());
-    }
-
-    /**
-     * Creates a new BlockPosition from given x, y, z coordinates.
-     */
-    public static Object createBlockPosition(int x, int y, int z) {
-        return blockPositionReflection.newInstance(
-                ClassObject.of(int.class, x),
-                ClassObject.of(int.class, y),
-                ClassObject.of(int.class, z)
-        );
+    public static TileEntity getTileEntity(Location loc) {
+        var world = (World) craftWorldReflection.invoke(loc.getWorld(), "getHandle");
+        return world.getTileEntity(new BlockPosition(loc.getBlockX(), loc.getBlockY(), loc.getBlockZ()));
     }
 
     /**
@@ -87,9 +61,9 @@ public class BlockUtils {
      * Attempts to retrieve the other half of a block.
      */
     public static Optional<Block> getOtherHalf(Block block) {
-        BlockData data = block.getBlockData();
+        var data = block.getBlockData();
         if (data instanceof Bed) {
-            Bed bed = (Bed) data;
+            var bed = (Bed) data;
             BlockFace facing = bed.getFacing();
             return Optional.of(block.getRelative(bed.getPart() == Bed.Part.HEAD ? facing.getOppositeFace() : facing));
         }

